src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1129,6 +1129,15 @@ private uint getMethodAttribsInternal(MethodDesc method)
             {
                 result |= CorInfoFlag.CORINFO_FLG_FORCEINLINE;
             }
+#if !READYTORUN
```

Comment:
Does it need this `#if !READYTORUN` ?

Reply:
It doesn't _need_ it, but it's a deoptimization if someone actually calls `Main`. It is valid to call `Main` from user code in .NET (as opposed to e.g. C++, where the language spec disallows it). I don't insist on keeping the ifdefs though...

Reply:
We can mark `MainMethodWrapper` as NoOptimization instead. It would not required ifdef and it would have a bit better fidelity with JIT behavior - Main would stay inlineable.

https://github.com/dotnet/runtime/blob/021ecb8eb7998bf3288ee68b12e952f775828c10/src/coreclr/tools/aot/ILCompiler.Compiler/IL/Stubs/StartupCode/StartupCodeMainMethod.cs#L207